graphql.md

```diff
@@ -0,0 +1,177 @@
+# GraphQL
+
+Mutations:
+
+- [Name should begin with action type](#name-should-begin-with-action-type)
+- [Use an input object type for the argument](#use-an-input-object-type-for-the-argument)
+- [Return an object in the payload](#return-an-object-in-the-payload)
+- [Errors](#errors)
+
+## Mutations
+
+Schema in the following examples
+
+```
+type User {
+  firstName: String!
+  lastName: String!
+  age: Int!
+}
+```
+
+### Name should begin with action type
+
+✅
+```gql
+
+mutation CreateUserMutation($input: UserInputType!) {
+   createUser(input: $input) {
+       user {
+           firstName
+       }
+   }
+}
+```
+
+✅
+```gql
+
+mutation DeleteUserMutation($input: UserInputType!) {
+   deleteUser(input: $input) {
+       user {
+           firstName
+       }
+   }
+}
+```
+
+### Use an input object type for the argument
+
+✅
+```gql
+
+mutation UpdateUserMutation($input: UserInputType!) {
+   updateUser(input: $input) {
+       user {
+           firstName
+       }
+   }
+}
+```
+
+where 
+
+```
+type UserInputType {
+    firstName: String!
+    lastName: String!
+    age: Int!
+}
+
+```
+
+❌
+```gql
+mutation UpdateUserMutation($firstName: String!, $lastName: String!, $age: Int!) {
+   updateUser(firstName: $firstName, lastName: $lastName, age: $age) {
+       user {
+           firstName
+       }
+   }
+}
+```
+
+Otherwise the mutations become more cumbersome to write out as the number of arguments grow
+
+Input object types are more flexible to changes. If the arguments change, less places need to be updated
+
+In the case where different mutations have the same arguments, the input object type can be reused.
+
+```gql
+mutation CreateUserMutation($input: UserInputType!) {
+   createUser(input: $input) {
+       user {
+           firstName
+       }
+   }
+}
+```
+
+### Return an object in the payload
+
+Ideally the object you've mutated
```

Comment:
Good to include a reason for each of these. For this one I'd go with - [so we can do magic cache busting](https://www.freecodecamp.org/news/how-to-update-the-apollo-clients-cache-after-a-mutation-79a0df79b840/)